File: linda-presentation/src/main/java/it/unibo/coordination/linda/presentation/RegularMatchSerializer.java
package it.unibo.coordination.linda.presentation;

import com.fasterxml.jackson.databind.ObjectMapper;
import it.unibo.coordination.linda.text.RegularMatch;
import it.unibo.coordination.linda.text.StringTuple;

import java.util.HashMap;

class RegularMatchSerializer extends DynamicSerializer<RegularMatch> {


    public RegularMatchSerializer(MIMETypes mimeType, ObjectMapper mapper) {
        super(mimeType, mapper);
    }

    @Override
    public Object toDynamicObject(RegularMatch object) {
        final var matchMap = new HashMap<String, Object>();

        matchMap.put("tuple", object.getTuple().map(StringTuple::getValue).orElse(null));
        matchMap.put("template", object.getTemplate().getTemplate().pattern());
        matchMap.put("match", object.isMatching());
        matchMap.put("map", object.toMap());

        return matchMap;
    }
}

File: linda-text-presentation/src/main/kotlin/it/unibo/coordination/linda/text/RegularMatchSerializer.kt
package it.unibo.coordination.linda.text

import com.fasterxml.jackson.databind.ObjectMapper
import it.unibo.presentation.DynamicSerializer
import it.unibo.presentation.MIMETypes
import java.util.*

internal class RegularMatchSerializer(mimeType: MIMETypes, mapper: ObjectMapper) : DynamicSerializer<RegularMatch>(mimeType, mapper) {
    override fun toDynamicObject(`object`: RegularMatch): Any {
        val matchMap: MutableMap<String, Any?> = HashMap()
        matchMap["tuple"] = `object`.tuple.map<String?> { obj: StringTuple -> obj.value }.orElse(null)
        matchMap["template"] = `object`.template.template.pattern()
        matchMap["match"] = `object`.isMatching
        matchMap["map"] = `object`.toMap()
        return matchMap
    }
}